tests/test_cookies.py

```diff
@@ -0,0 +1,501 @@
+    http = httplib2.Http(cookie_jar=cookie_jar)
+    yield http
+    http.close()
+
+
+SET_COOKIE_HEADERS = ("A=1; Max-Age=100,B=2, C=3", ("A=1; Max-Age=100", "B=2", "C=3"))
+
+
+@pytest.mark.parametrize("cookie_header", SET_COOKIE_HEADERS, ids=["single", "multiple"])
+def test_response_cookie_attribute(cookie_header):
+    """ Test the attribute 'cookies' in Response object."""
+    http = httplib2.Http()
+
+    with tests.server_const_http(headers={"set-cookie": cookie_header}) as uri:
+        response, _ = http.request(uri, "GET")
+        assert len(response.cookies) == 3
+        assert response.cookies["A"] == "1"
+        assert response.cookies["B"] == "2"
+        assert response.cookies["C"] == "3"
+
+
+def test_onetime_cookie():
+    http = httplib2.Http()
+    cookies = {"AAA": "111", "BBB": "222"}
+    handler = tests.http_reflect_with_cookies()
+
+    with tests.server_request(handler) as uri:
+        response, content = http.request(uri, "GET", body="this request have one time cookie", cookies=cookies)
+        assert response.status == 200
+        assert dict([ck.split("=") for ck in content.decode().split("; ")]) == cookies
+
+
+@pytest.mark.parametrize("cookie_header", SET_COOKIE_HEADERS, ids=["single", "multiple"])
+def test_cookie_manager_enabled(cookie_header, cookie_http):
+    print(cookie_header)
+    handler = tests.http_reflect_with_cookies({"/login": cookie_header})
+
+    with tests.server_request(handler, request_count=2) as uri:
+        login_url = urllib.parse.urljoin(uri, "/login")
+        response, content = cookie_http.request(login_url, "POST", body="this is a login request")
+        assert response.status == 200
+        assert len(response.cookies) == 3
+
+        op_url = urllib.parse.urljoin(uri, "/op")
+        response, content = cookie_http.request(op_url, "POST", body="this is an op request")
+        assert response.status == 200
+
+        if isinstance(cookie_header, (list, tuple)):
+            cookie_header = ", ".join(cookie_header)
+        set_cookies = {ck["name"]: ck["value"] for ck in httplib2.CookieJar.parse_iter(cookie_header)}
+        assert dict([ck.split("=") for ck in content.decode().split("; ")]) == set_cookies
+
+
+def test_cookie_manager_expire(cookie_http):
+    expires = (datetime.utcnow() + timedelta(seconds=10)).strftime("%a, %d %b %Y %H:%M:%S GMT")
+    cookie_header = ("A=1", "B=2; Max-Age=5", "C=3; Expires={}".format(expires))
+    handler = tests.http_reflect_with_cookies({"/login": cookie_header})
+
+    with tests.server_request(handler, request_count=4, timeout=20) as uri:
+        login_url = urllib.parse.urljoin(uri, "/login")
+        response, content = cookie_http.request(login_url, "POST", body="this is a login request")
+        assert response.status == 200
+
+        op_url = urllib.parse.urljoin(uri, "/op")
+        response, content = cookie_http.request(op_url, "POST", body="op request 1")
+        assert response.status == 200
+
+        time.sleep(5)
```

Comment:
```suggestion
        time.sleep(1)  # FIXME patch time
```

Reply:
There is a reason why I don't use a smaller expire time.  Because test server doesn't listen on IPv6 address.  But each http.request() will try to the [::1] first until 2 seconds timeout, then, it will connect to 127.0.0.1 to complete the session. Thus, a smaller expire time could cause the cookie expired during the first request.
![image](https://user-images.githubusercontent.com/5524708/99756635-90070f00-2b28-11eb-8ba1-ec1abf3cbcf0.png)

The solution is modify the `server_socket` to listen on both IPv4 and IPv6 address:
   
```
def server_socket(...)
...
        - server = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
        + server = socket.socket(socket.AF_INET6, socket.SOCK_STREAM)
        + server.setsockopt(socket.IPPROTO_IPV6, socket.IPV6_V6ONLY, 0)
```

What's your advise?

Reply:
I would modify time and remove all sleeps. It may require exposing cookie time source like so

```
# py{2,3}/httplib2/cookie.py
now = time.now

def fun():
-    time.now()
+    now()
```

Reply:
It looks like there is no no problem on CI check with 1 second expire time.